components/patient-data/indexing/src/main/java/org/phenotips/data/indexing/internal/SolrPatientIndexer.java

```diff
@@ -138,4 +165,31 @@ public void reindex()
             this.logger.warn("Failed to search patients for reindexing: {}", ex.getMessage());
         }
     }
+
+    private void addGenes(SolrInputDocument input, PatientData<Map<String, String>> allGenes)
+    {
+        if (allGenes != null && allGenes.isIndexed()) {
+            for (Map<String, String> gene : allGenes) {
+                String name = gene.get(GENE_NAME_FIELD);
+                if (StringUtils.isBlank(name)) {
+                    continue;
+                }
+
+                String geneStatus = gene.get(GENE_STATUS_FIELD);
+                String solrField = null;
+                // Index genes with empty or null status as candidates
+                if (StringUtils.isBlank(geneStatus) || GENE_STATUS_CANDIDATE.equals(geneStatus)) {
+                    solrField = SOLR_FIELD_CANDIDATE_GENES;
+                } else if (GENE_STATUS_SOLVED.equals(geneStatus)) {
+                    solrField = SOLR_FIELD_SOLVED_GENES;
+                } else if (GENE_STATUS_REJECTED.equals(geneStatus)) {
+                    solrField = SOLR_FIELD_REJECTED_GENES;
+                }
+
+                if (solrField != null) {
+                    input.addField(solrField, name);
```

Comment:
might be overthinking, but as alternative `geneStatus = StringUtils.isBlank(geneStatus) ? GENE_STATUS_CANDIDATE : geneStatus` and `input.addField(geneStatus+"_genes", name)` instead of `if-else` chain and many static vars. Came from that `solrField` seems to be never `null`